zanata-war/src/main/java/org/zanata/action/ZanataRoleAction.java

```diff
@@ -0,0 +1,131 @@
+package org.zanata.action;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jboss.seam.annotations.Begin;
+import org.jboss.seam.annotations.Create;
+import org.jboss.seam.annotations.In;
+import org.jboss.seam.annotations.Install;
+import org.jboss.seam.annotations.Name;
+import org.jboss.seam.annotations.Scope;
+import org.jboss.seam.core.Conversation;
+import org.zanata.security.ZanataIdentity;
+import org.zanata.seam.security.ZanataJpaIdentityStore;
+import org.zanata.security.annotations.CheckLoggedIn;
+import org.zanata.security.annotations.CheckRole;
+import org.zanata.security.annotations.ZanataSecured;
+
+import static org.jboss.seam.ScopeType.CONVERSATION;
+import static org.jboss.seam.annotations.Install.APPLICATION;
+
+/**
+ * @author Patrick Huang
+ *         <a href="mailto:pahuang@redhat.com">pahuang@redhat.com</a>
+ */
+@Name("zanataRoleAction")
+@Scope(CONVERSATION)
+@Install(precedence = APPLICATION)
+@ZanataSecured
+@CheckLoggedIn
+@CheckRole("admin")
+public class ZanataRoleAction implements Serializable {
```

Comment:
RoleAction is used by roledetail.xhtml and rolemanager.xhtml. They are both under admin/ which were restricted to admin only. Same reason for UserAction. 

----- Original Message -----

> From: "Sean Flanigan" <notifications@github.com>
> To: "zanata/zanata-server" <zanata-server@noreply.github.com>
> Cc: "Patrick Huang" <pahuang@redhat.com>
> Sent: Friday, July 31, 2015 3:08:26 PM
> Subject: Re: [zanata-server] Replace Seam Security (#908)

> In zanata-war/src/main/java/org/zanata/action/ZanataRoleAction.java :
> > +import org.zanata.security.annotations.ZanataSecured;
> > +
> > +import static org.jboss.seam.ScopeType.CONVERSATION;
> > +import static org.jboss.seam.annotations.Install.APPLICATION;
> > +
> > +/**
> > + * @author Patrick Huang
> > + *         <a href="mailto:pahuang@redhat.com">pahuang@redhat.com</a>
> > + */
> > +@Name("zanataRoleAction")
> > +@Scope(CONVERSATION)
> > +@Install(precedence = APPLICATION)
> > +@ZanataSecured
> > +@CheckLoggedIn
> > +@CheckRole("admin")
> > +public class ZanataRoleAction implements Serializable {

> Seam's RoleAction wasn't restricted to admin. Why is ZanataRoleAction?

> —
> Reply to this email directly or view it on GitHub .